inbound_internal_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tchannel
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/opentracing/opentracing-go/mocktracer"
+)
+
+type statsReporter struct{}
+
+func (w *statsReporter) IncCounter(name string, tags map[string]string, value int64) {
+}
+
+func (w *statsReporter) UpdateGauge(name string, tags map[string]string, value int64) {
+}
+
+func (w *statsReporter) RecordTimer(name string, tags map[string]string, d time.Duration) {
+}
+
+func TestTracingSpanApplicationError(t *testing.T) {
```

Comment:
can you merge this test and the one below as a single table-driven test?

separately, can you add an end-to-end test for this? testing the implementation is great, but end-to-end tests are important since this library is at a very low layer of the stack.

Reply:
I was able to address all the comments for the implementation specific tests, however, I could not figure out a reliable way of writing an integration test to cover this flow (nothing similar already exists)